server/src/main/java/io/crate/lucene/LuceneQueryBuilder.java

```diff
@@ -203,13 +203,8 @@ public SymbolVisitor<Context, Query> visitor() {
         }
 
         @Nullable
-        public Reference getRef(String field) {
-            try {
-                long oid = Long.parseLong(field);
-                return table.getReference(oid);
-            } catch (NumberFormatException ex) {
-                return getRef(ColumnIdent.fromPath(field));
-            }
+        public Reference getRef(String storageIdent) {
+            return table.getReference(storageIdent);
```

Comment:
I wonder if can rely on version.created instead of catching an exception.

Reply:
OK, looks like version wouldn't helps for all cases but we already have `lookupNameBySourceKey`?

Reply:
Yes I'm also not happy about the `O(n)` lookup but I'd like to keep it for now and follow up on it later to unblock the other work that builds on the changes here.